web-ui/src/main/java/org/archcnl/ui/input/ListEntry.java

```diff
@@ -0,0 +1,57 @@
+package org.archcnl.ui.input;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.archcnl.domain.input.datatypes.mappings.Concept;
+import org.archcnl.domain.input.datatypes.mappings.Relation;
+
+public class ListEntry<T> {
+
+    private List<T> children;
+    private String name;
+    private T content;
+    private boolean isLeaf;
+
+    public ListEntry(String name, List<T> children) {
+        this.children = children;
+        this.name = name;
+        isLeaf = false;
+    }
+
+    public ListEntry(String name, T content) {
+        this.name = name;
+        this.content = content;
+        children = Collections.<T>emptyList();
+        isLeaf = true;
+    }
+
+    public T getContent() {
+        return content;
+    }
+
+    public List<ListEntry<T>> getHierarchicalChildren() {
+        List<ListEntry<T>> hierarchicalChildren =
+                children.stream()
+                        .map(entry -> (ListEntry<T>) new ListEntry<>(entry.toString(), entry))
+                        .collect(Collectors.toList());
+        return hierarchicalChildren;
+    }
+
+    // temporal solution, ideally Concept and Relation should have these methods as the toString
+    // method
+    @Override
+    public String toString() {
```

Comment:
Maybe extract the different behavior for Concept and Relation into subclasses and make this class abstract

Reply:
If toString will be overwritten in Concept and Relation I think leaving it, as it is looks fine to me, as then there would be no specifics needed inside this class for now. If it will not be overwritten or there is some other functionality that differs between them I can still change it if you want.

Reply:
Then we can leave it for now until the getName()/toString() stuff has been changed